src/include/sof/audio/kpb.h

```diff
@@ -150,6 +159,16 @@ struct history_data {
 	struct history_buffer *c_hb; /**< current buffer used for writing */
 };
 
+enum ipc4_kpb_module_config_params {
+	    /* Mic selector for client - sets microphone id for real time sink mic selector */
+	    KP_BUF_CLIENT_MIC_SELECT = 11,
```

Comment:
Why 11 ? if its for Windows compatibility we should say so that no one changes it later.

Reply:
For compatibility with IPC4 and tests:
enum KpbModuleConfigParams
{
    //! Configure the buffer size for Keyphrase,onset and buffer to skip sizes.
    KP_BUF_CFG_MODE   = 0,
    //! Configure the module ID's which would be part of the Fast mode tasks
    KP_BUF_CFG_FM_MODULE  = 1,
    //! Parameter to retrieve KP position in buffer
    /*!
      Use LARGE_CONFIG_GET to retrieve KP buffer position in bytes of samples
      for kpb module inout format.

      \see KeyPhraseBufferPosition
    */
    KP_BUF_POS_IN_BUFFER = 2,
    //! Parameter to request Draining On Demand (DOD).
    /*!
      Start draining given pin without KPB.
    */
    KP_BUF_PIN_DOD = 3,
    //! Parameter to retrieve KP position in buffer
    /*!
      Use LARGE_CONFIG_GET to retrieve KP buffer position in bytes of samples
      for kpb module inout format.

      \see KeyPhraseBufferPositionExtended

      Sample code to retrieve KeyPhraseBufferPositionExtended:
      \code
      Message::LargeConfigOp message(true, KPB_MODULE_ID, KPB_INSTANCE_ID);
      message.GetBits().large_param_id = Message::LargeConfigOp::VENDOR_CONFIG_PARAM;
      message.GetBits().init_block = true;
      message.GetBits().final_block = true;
      message.GetBits().data_off_size = sizeof(TL); // type/length struct

      Message::LargeConfigOp::ExtendedParamId param = {0};
      param.part.parameter_type = KP_BUF_POS_IN_BUFFER_EXTENDED;
      param.part.parameter_instance = MODEL_ID; // KPB_CLIENT_ID
      TL input_mailbox(param.full, sizeof(KeyPhraseBufferPositionExtended));

      KeyPhraseBufferPositionExtended* output_mailbox = NULL;
      send_ipc(message, input_mailbox, (uint8_t**)&output_mailbox);

      \endcode
    */
    KP_BUF_POS_IN_BUFFER_EXTENDED = 4,
    KP_BUF_GET_DATA = 7,
    //! Configure the module to enable ULP support
    /*!
      The command enables ULP support / ULP Audio Gateway support in
      History Buffer.
    */
    KP_BUF_ULP_SUPPORT = 8,
    //! Specifies all data consumers attached to History Buffer.
    /*!
        It is also needed when a module wants to request data to be streamed
        via DATA_CONSUMER_REGISTERED and DATA_CONSUMER_UNREGISTERED messages.
    */
    KP_BUF_DATA_CONSUMER_INFO = 9,
    /*! Sent when a reference signal from playback is connected/disconnected to/from
        the capture path
     */
    KP_BUF_REFERENCE_DATA_STATE = 10,
    //! Mic selector for client - sets microphone id for real time sink mic selector
    KP_BUF_CLIENT_MIC_SELECT = 11,
};

Reply:
Comment added to code